shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Mempool.hs

```diff
@@ -61,6 +63,15 @@ import Shelley.Spec.Ledger.PParams (PParams' (..))
 import Shelley.Spec.Ledger.STS.Ledger (LedgerEnv, LedgerPredicateFailure)
 import qualified Shelley.Spec.Ledger.STS.Ledger as Ledger
 
+-- | A newtype which indicates that a transaction has been validated against
+-- some chain state.
+newtype Validated tx = Validated tx
```

Comment:
I need a `NoThunks` instance for this.

Reply:
And also a `TranslateEra` instance for it.

I might be able to get by without this, with a little refactoring on our end. But at the moment, we use types to ensure that the various forge functions only receive validated transactions. The forge functions themselves, though, immediately "forget" the validation newtype wrapper -- so I think could weaken our internal types in order to avoid needing to translate validated txs if you're uncomfortable adding that.